app/scripts/modules/core/src/pipeline/config/stages/monitorPipeline/MonitorPipelineStageExecutionDetails.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { UISref } from '@uirouter/react';
+
+import { IExecutionDetailsSectionProps, ExecutionDetailsSection } from '../common';
+import { StageFailureMessage } from '../../../details';
+import _ from 'lodash';
+
+export function MonitorPipelineStageExecutionDetails(props: IExecutionDetailsSectionProps) {
+  const {
+    application,
+    execution,
+    stage: { context = {}, outputs = {} },
+    stage,
+    name,
+    current,
+  } = props;
+
+  const statuses = Object.keys(outputs.executionStatuses).map((executionId) => {
+    const status = outputs.executionStatuses[executionId];
```

Comment:
We should avoid mutating executions.


```suggestion
    const status = {...outputs.executionStatuses[executionId]};
```